src/rosdep2/sources_list.py

```diff
@@ -320,9 +321,12 @@ def download_default_sources_list(url=DEFAULT_SOURCES_LIST_URL):
     data = f.read().decode()
     f.close()
     if not data:
-        raise RuntimeError("cannot download defaults file: empty contents")
+        raise DownloadFailure("cannot download defaults file: empty contents")
```

Comment:
It might be more helpful if the exception raised would contain the `url` for which the problem occured.